tensorflow_addons/optimizers/moving_average_test.py

```diff
@@ -15,79 +15,75 @@
 """Tests for MovingAverage optimizers."""
 
 import numpy as np
+import pytest
 import tensorflow as tf
 
 from tensorflow_addons.optimizers import MovingAverage
 from tensorflow_addons.utils import test_utils
 
 
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("sequential_update", [True, False])
+def test_run(sequential_update):
+    var0 = tf.Variable([1.0, 2.0])
+    var1 = tf.Variable([3.0, 4.0])
+
+    grads0 = tf.constant([0.1, 0.1])
+    grads1 = tf.constant([0.01, 0.01])
+
+    grads_and_vars = list(zip([grads0, grads1], [var0, var1]))
+
+    opt = MovingAverage(
+        tf.keras.optimizers.SGD(lr=2.0),
+        sequential_update=sequential_update,
+        average_decay=0.5,
+    )
+
+    opt.apply_gradients(grads_and_vars)
+    opt.apply_gradients(grads_and_vars)
+
+    np.testing.assert_allclose(var0.read_value(), [0.6, 1.6])
+    np.testing.assert_allclose(var1.read_value(), [2.96, 3.96])
+
+    ema_var0 = opt.get_slot(var0, "average")
+    ema_var1 = opt.get_slot(var1, "average")
+
+    if sequential_update:
+        np.testing.assert_allclose(ema_var0.read_value(), [0.75, 1.75])
+        np.testing.assert_allclose(ema_var1.read_value(), [2.975, 3.975])
+
+    _ = opt.assign_average_vars([var0, var1])
+
+    if sequential_update:
+        np.testing.assert_allclose(var0.read_value(), [0.75, 1.75])
+        np.testing.assert_allclose(var1.read_value(), [2.975, 3.975])
+
+    _ = tf.group(
+        [
+            var0.assign_add([1.0, 1.0]),
+            var1.assign_add([2.0, 2.0]),
+            ema_var0.assign_add([3.0, 3.0]),
+            ema_var1.assign_add([4.0, 4.0]),
+        ]
+    )
```

Comment:
```suggestion
    var0.assign_add([1.0, 1.0])
    var1.assign_add([2.0, 2.0])
    ema_var0.assign_add([3.0, 3.0])
    ema_var1.assign_add([4.0, 4.0])
```

In eager, the assign_add operation is executed immediately. No need to use a group and then a evaluate.

Reply:
Thanks @gabrieldemarmiesse